### PR TITLE
add instructions for pinning packages

### PR DESCRIPTION
**Pull request by cfcs**
This updates the README with instructions for pinning the various packages given by Hannes in https://github.com/roburio/udns/issues/28

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,16 +75,67 @@ The µDNS library is published under the 2 clause BSD license.
...
+You may want to follow the [mirage installation
 instructions](https://mirage.io/wiki/install) to get `mirage` installed on your
 computer.
 
-µDNS is not released yet, but you can install it and its dependencies via opam:
-`opam pin add dns https://github.com/roburio/udns.git`
+To lower the amount of run-time dependencies for each individual functionality,
+the library is split across a number of opam packages.
+
+µDNS is not released yet, but you can install it and its dependencies via opam, see [Development](#Development).
```
Comment (hannesm) [inline README.md]: this used to be 80 characters wide, now it is no longer... would you mind inserting a linebreak after the `,`?

**Comment by hannesm** [comment]
looks good to me apart from the minor nit -- also CI failure looks like a missing unix dependency for `ohost.exe`:

```bash
#=== ERROR while compiling dns-client-unix.dev ================================#
# context              2.0.4 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(file:///repo)
# path                 ~/.opam/4.07/.opam-switch/build/dns-client-unix.dev
# command              ~/.opam/4.07/bin/dune build -p dns-client-unix -j 71
# exit-code            1
# env-file             ~/.opam/log/dns-client-unix-13749-d2c37b.env
# output-file          ~/.opam/log/dns-client-unix-13749-d2c37b.out
### output ###
#     ocamlopt unix/client/ohost.exe (exit 2)
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamlopt.opt -w -40 -g -o unix/client/ohost.exe -I /home/opam/.opam/4.07/lib/astring -I /home/opam/.opam/4.07/lib/bigarray-compat -I /home/opam/.opam/4.07/lib/bytes -I /home/opam/.opam/4.07/lib/cstruct -I /home/opam/.opam/4.07/lib/dns -I /home/opam/.opam/4.07/lib/dns-client -I /home/opam/.opam/4.07/lib/domain-name -I /home/opam/.opam/4.07/lib/fm[...]
# File "_none_", line 1:
# Error: No implementations provided for the following modules:
#          Unix referenced from unix/client/dns_client_unix.cmxa(Dns_client_unix)
```

**Comment by cfcs** [comment]
Done :)

**Comment by hannesm** [comment]
ths, merged as part of #32

**Pull request closed without merging**